src/charm.py

```diff
@@ -418,13 +420,21 @@ def _create_db(self, event, db_name, user):
             conn.execute(f"GRANT ALL PERMISSIONS ON KEYSPACE {db_name} to '{user}'")
 
     def _configure(self, event):
+        heap_size = self.model.config["heap_size"]
+
+        if match("^\\d+[kKmMgG]$", heap_size) is None:
```

Comment:
If you want to allow floating point values shouldn't the regular expression be something like `\d+(\.\d*)` . Also if you use raw strings then backslashes will not need to be escaped.

Nitpick: strictly speaking lower case "k" "m" and "g" are not standard memory sizes.

Reply:
Yeh, I'd expect `G`, `M`, `K` or `GiB`, `MiB` etc.

Reply:
I think (reading the doc string above) that the intention is to explicitly not support decimal/float value here.

```suggestion
        if match(r"^\d+[KMG]$", heap_size) is None:
```

Reply:
@balbirthomas the JVM accepts only integers. That is why I am testing for failure with `0.5g` :-)